### PR TITLE
[RFR] New transformers

### PR DESCRIPTION
**Pull request by touv**
- trim
- capitalize
- uniq

### CODE REVIEW

File: src/common/transformers/TRIM.spec.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    });
+
+    it('should return trim value of each item in array', () => {
+        expect(trimString([' hello   ', '\nworld\r\n'])).toEqual([
+            'hello',
+            'world',
+        ]);
+    });
+
+    it('should return null if value is literal', () => {
```
Comment (parmentf) [inline src/common/transformers/TRIM.spec.js]: if value is NOT a literal?

File: src/common/transformers/UNIQ.spec.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    });
+
+    it('should return uniq value of each item in array', () => {
+        expect(uniqArray(['hello', 'world', 'hello', 'world'])).toEqual([
+            'hello',
+            'world',
+        ]);
+    });
+
+    it('should return null if value is literal', () => {
```
Comment (parmentf) [inline src/common/transformers/UNIQ.spec.js]: if value is NOT a literal, or better: if value is an object.

Comment (touv) [reply src/common/transformers/UNIQ.spec.js]: I didn't change text from uppercase.spec.js, it's a copy/paste

File: src/common/transformers/TRIM.spec.js (hunk 1 above)
Comment (ThieryMichel) [reply src/common/transformers/TRIM.spec.js]: no, the text is right, { a: 'hello ', b: 'world ' } is a literal. But maybe `is not a string` would be even better.

File: src/common/transformers/UNIQ.spec.js (hunk 2 above)
Comment (ThieryMichel) [reply src/common/transformers/UNIQ.spec.js]: The text is right here too. And is `not an array` is even better

File: src/common/transformers/UNIQ.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
+export const uniqArray = value => {
+    if (!value) {
+        return null;
+    }
+
+    if (Array.isArray(value)) {
+        return value.filter((v, i, a) => a.indexOf(v) === i);
```
Comment (ThieryMichel) [inline src/common/transformers/UNIQ.js]: You should use lodash.uniq instead, it do the same and is bullet proof.

File: src/common/transformers/CAPITALIZE.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+export const capitalizeString = value => {
+    if (!value) {
+        return null;
+    }
+
+    if (Array.isArray(value)) {
+        return value.map(capitalizeString);
+    }
+
+    if (typeof value === 'object') {
```
Comment (ThieryMichel) [inline src/common/transformers/CAPITALIZE.js]: A suggstion: wouldn't it better if you capitalized the value of the literal ?
```js
Object.keys(value).reduce((acc, key) => ({
    ...acc,
    [key]: capitalizeString(value[key]),
}));
```

File: src/common/transformers/TRIM.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+export const trimString = value => {
+    if (!value) {
+        return null;
+    }
+
+    if (Array.isArray(value)) {
+        return value.map(trimString);
+    }
+
+    if (typeof value === 'object') {
```
Comment (ThieryMichel) [inline src/common/transformers/TRIM.js]: Suggestion: you could trim value of literal (see capitalize)

File: src/common/transformers/CAPITALIZE.js [hunk 6]
```diff
@@ -0,0 +1,31 @@
+export const capitalizeString = value => {
+    if (!value) {
```
Comment (ThieryMichel) [inline src/common/transformers/CAPITALIZE.js]: Be it capitalize, uniq, or trim, the only thing that change is the function applied to the value.
Please refactor in order to reuse the common logic.

eg:

```js
const transform = fn => value => {
    if(!value) {
        return null;
    }
     ...
    return fn(value);
}
```

Comment (ThieryMichel) [reply src/common/transformers/CAPITALIZE.js]: This way you only have to test transform, and fn, and this will avoid all test copy paste.

**Review by ThieryMichel — CHANGES_REQUESTED** [review]

**Review by ThieryMichel — APPROVED** [review]

**Pull request merged**